src/octopeer-github/main/DatabaseAdapter.ts

```diff
@@ -0,0 +1,91 @@
+class DatabaseAdapter {
+
+    private _url: string;
+    private _session: number;
+    private _pr: number;
+    private _user: number;
+    private _initialized: boolean;
```

Comment:
There is more state in this Adapter, namely the session number, pr number, etc. I intended the _initialized variable to make sure POST calls can only be made when the Adapter is correctly initialized (which implies: it has a connection to the server).